ingest/src/test/java/com/microsoft/azure/kusto/ingest/ManagedStreamingIngestClientTest.java

```diff
@@ -0,0 +1,344 @@
+        when(resultSet.getObject(1)).thenReturn("Name");
+        when(resultSet.getObject(2)).thenReturn("Age");
+        when(resultSet.getObject(3)).thenReturn("Weight");
+
+        when(resultSetMetaData.getColumnCount()).thenReturn(3);
+
+        ArgumentCaptor<InputStream> argumentCaptor = ArgumentCaptor.forClass(InputStream.class);
+
+        ResultSetSourceInfo resultSetSourceInfo = new ResultSetSourceInfo(resultSet);
+        OperationStatus status = managedStreamingIngestClient.ingestFromResultSet(resultSetSourceInfo, ingestionProperties).getIngestionStatusCollection().get(0).status;
+        assertEquals(status, OperationStatus.Succeeded);
+        verify(streamingClientMock, atLeastOnce()).executeStreamingIngest(any(String.class), any(String.class), argumentCaptor.capture(),
+                isNull(), any(String.class), eq("mappingName"), any(boolean.class));
+
+        InputStream stream = argumentCaptor.getValue();
+        verifyCompressedStreamContent(stream, "Name,Age,Weight");
+    }
+
+    @Test
+    void IngestFromFile_Csv() throws Exception {
+        String resourcesDirectory = System.getProperty("user.dir") + "/src/test/resources/";
+        String path = resourcesDirectory + "testdata.csv";
+        FileSourceInfo fileSourceInfo = new FileSourceInfo(path, new File(path).length());
+        OperationStatus status = managedStreamingIngestClient.ingestFromFile(fileSourceInfo, ingestionProperties).getIngestionStatusCollection().get(0).status;
+        assertEquals(status, OperationStatus.Succeeded);
+        verify(streamingClientMock, atLeastOnce()).executeStreamingIngest(any(String.class), any(String.class), any(InputStream.class),
+                isNull(), any(String.class), eq("mappingName"), any(boolean.class));
+    }
+
+    @Test
+    void IngestFromFile_Json() throws Exception {
+        String resourcesDirectory = System.getProperty("user.dir") + "/src/test/resources/";
+        String path = resourcesDirectory + "testdata.json";
+        FileSourceInfo fileSourceInfo = new FileSourceInfo(path, new File(path).length());
+        ingestionProperties.setDataFormat(IngestionProperties.DATA_FORMAT.json);
+        ingestionProperties.setIngestionMapping("JsonMapping", IngestionMapping.IngestionMappingKind.Json);
+        OperationStatus status = managedStreamingIngestClient.ingestFromFile(fileSourceInfo, ingestionProperties).getIngestionStatusCollection().get(0).status;
+        assertEquals(status, OperationStatus.Succeeded);
+        verify(streamingClientMock, atLeastOnce()).executeStreamingIngest(any(String.class), any(String.class), any(InputStream.class),
+                isNull(), any(String.class), any(String.class), any(boolean.class));
+    }
+
+    @Test
+    void IngestFromFile_CompressedJson() throws Exception {
+        String resourcesDirectory = System.getProperty("user.dir") + "/src/test/resources/";
+        String path = resourcesDirectory + "testdata.json";
```

Comment:
It's the same path you use for the prev test. although from the method name you should test here a compressed json

Reply:
You're right, since that's also a problem in StreamingIngestClientTest.java I fixed it in both now